model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.utils
+
+import com.here.ort.model.Identifier
+import com.here.ort.model.LicenseFindings
+import com.here.ort.model.OrtResult
+import com.here.ort.model.config.PathExclude
+
+import java.util.SortedSet
+
+/**
+ * Return a map of license findings for each project or package [Identifier]. The license findings for projects are
+ * mapped to a list of [PathExclude]s matching the locations where a license was found. This list is only populated
+ * if all file locations are excluded. The list is empty for all dependency packages, as path excludes are only
+ * applied to the projects.
+ *
+ * If [omitExcluded] is set to true, excluded projects / packages are omitted from the result.
+ */
+fun OrtResult.collectLicenseFindings(
```

Comment:
I'd appreciate if it was clearly defined which functions from `OrtResult` are now moved to this file as extension functions, and which are not, as currently the selection seems quite arbitrary.

My proposal would be to move all function that

- have `@Suppress("UNUSED")` as they are somewhat special and only used from scripts.
- have `@JsonIgnore` (i.e. all getters that take no parameters), because as extension functions we could drop the `@JsonIgnore` I believe.

Maybe it even makes sense to put all scripting functions to a separate `(OrtResult)ScriptingExtension.kt` (or similar) file, and only put the other functions into `OrtResultExtensions.kt`.

Reply:
The selection with `not arbitrary` at all as I moved `collectLicenseFindings()` along with its callers which are also related as they deal with `detected` licenses. Anyhow, I believe moving additional functions to that file can be nicely done incrementally.
Anyhow I believe moving further functions is out of scope, can we agree on that?

Reply:
> The selection was not made `arbitrary` at all as I moved `collectLicenseFindings()` along with its callers

Yes, I got that from the commit message. But this is a very narrow scope (which is not bad per se), and you move these to a generically named `OrtResultExtensions.kt` file, which can be confusing without more context.

As such I would have expected the functions to either be also moved to `LicenseResolver.kt`, **or** explain that you are expecting more functions to be moved to `OrtResultExtensions.kt` to justify its creation and generic name.

> Most importantly, I believe moving further functions is out of scope, can we agree on that?

Yes, I'll happily do the cleanup later by moving more fitting functions to `OrtResultExtensions.kt`. I wanted to have that file already some time ago, back when you still objected to moving out functions from `OrtResult` to extension functions :wink:

Reply:
Regarding the `ScriptingExtension.kt` I think the underlying idea is to group the function by (1) the place from where they are called instead of (2) grouping related functions together.

I see the following problems with  (1)
* it has less cohession
* can require to move functions back and forth if the caller changes
* hard to tell in advance if the function is called only from scripting
* The functions use from scripting are not located at a single place as in case there is at least one caller from within ORT they would be moved out of the `ScriptingExtension.kt`

What do you think about that?

Reply:
My intention is indeed to group *related* functions. But sometimes, functions are related because they have something in common, and that "something" could be their intended use e.g. exclusively as script functions.

But I agree that this could cause moving functions back and forth, so let's scratch the idea of a separate `ScriptingExtension.kt`.

Reply:
> I wanted to have that file already some time ago, back when you still objected to moving out functions from OrtResult to extension function

yes, I recalled that which is why I started this file now ;-).